torch_xla/experimental/tpu.py

```diff
@@ -81,11 +81,26 @@ def task_id() -> Optional[int]:
   return xu.getenv_as(xenv.CLOUD_TPU_TASK_ID, int)
 
 
+def build_tpu_env_from_vars() -> Dict[str, str]:
+  metadata = defaultdict(str)
+  metadata[xenv.ACCELERATOR_TYPE] = xu.getenv_as(xenv.TPU_ACCELERATOR_TYPE, str)
+  metadata[xenv.TPU_PROCESS_BOUNDS] = xu.getenv_as(
+      xenv.TPU_PROCESS_BOUNDS, str, xu.getenv_as(xenv.TPU_HOST_BOUNDS, str))
+  metadata[xenv.TPU_CHIPS_PER_PROCESS_BOUNDS] = xu.getenv_as(
+      xenv.TPU_CHIPS_PER_PROCESS_BOUNDS, str,
+      xu.getenv_as(xenv.TPU_CHIPS_PER_HOST_BOUNDS, str))
+  metadata[xenv.WORKER_ID] = xu.getenv_as(xenv.CLOUD_TPU_TASK_ID, str,
+                                          xu.getenv_as(xenv.TPU_WORKER_ID, str))
+  return metadata
+
+
 def get_tpu_env() -> Dict[str, str]:
   """Fetches and parses `tpu-env` metadata field."""
-  metadata = _get_metadata('tpu-env')
-
-  return yaml.load(metadata, yaml.Loader)
+  metadata = build_tpu_env_from_vars()
+  if metadata[xenv.ACCELERATOR_TYPE] is None:
```

Comment:
I think the logic here could be easier to follow, since this calls the env vars "metadata" and then conditionally ignores it and returns the actual metadata. Also, the first definition of `metadata` is a `dict` and the second is a `str`.

Let's work out the condition is for using the environment variables compared to metadata and put that up front, rather than having the logic broken up. e.g. something of the form

```
if using_env_vars():
  return build_tpu_env_from_vars()

# otherwise fetch GCE metadata
```

Also see my main comment for a design question.

Reply:
Done